### PR TITLE
photocollage: init at 1.4.5

### PR DESCRIPTION
**Pull request by Pamplemousse**
###### Description of changes

A Python application to build photo collage posters - https://github.com/adrienverge/PhotoCollage .

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/applications/graphics/photocollage/default.nix [hunk 1]
```diff
@@ -0,0 +1,47 @@
+{ lib
+, pkgs
+, python
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/photocollage/default.nix:3]:
```suggestion
, python3
```

File: pkgs/applications/graphics/photocollage/default.nix [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+, wrapGAppsHook
+}:
+
+with python.pkgs;
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/photocollage/default.nix:7]:
```suggestion
with python3.pkgs;
```

File: pkgs/top-level/all-packages.nix [hunk 3]
```diff
@@ -16140,6 +16140,9 @@ with pkgs;
 
   pachyderm = callPackage ../applications/networking/cluster/pachyderm { };
 
+  photocollage = callPackage ../applications/graphics/photocollage {
+    python = python3;
+  };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:16145]:
```suggestion
  photocollage = callPackage ../applications/graphics/photocollage { };
```

File: pkgs/applications/graphics/photocollage/default.nix [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    pygobject3
+  ];
+
+  patches = [
+    ./import_gi.patch
+  ];
+
+  postPatch = ''
+    substituteInPlace setup.py --replace "\"msgfmt\"" "\"${pkgs.gettext}/bin/msgfmt\""
+  '';
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/photocollage/default.nix:39]: Why not add gettext to nativeBuildInputs?

File: pkgs/applications/graphics/photocollage/default.nix [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+    pkgs.gettext
+    pkgs.gtk3
+
+    pillow
+    pycairo
+    pygobject3
+  ];
+
+  patches = [
+    ./import_gi.patch
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/photocollage/default.nix:34]: Should we upstream this?

File: pkgs/applications/graphics/photocollage/default.nix [hunk 6]
```diff
@@ -0,0 +1,47 @@
...
+
+  nativeBuildInputs = [
+    pkgs.gobject-introspection
+    wrapGAppsHook
+  ];
+
+  propagatedBuildInputs = [
+    pkgs.gdk-pixbuf
+    pkgs.gettext
+    pkgs.gtk3
```
Comment (SuperSandro2000) [inline pkgs/applications/graphics/photocollage/default.nix:26]: We are not in pythonPackages, so we can add them to the inputs quite easy

**Comment by doronbehar** [comment]
Closing in favor of #277980.

**Pull request closed without merging**